[debug] [2024-02-02T08:01:25.513Z] ----------------------------------------------------------------------
[debug] [2024-02-02T08:01:25.523Z] Command:       C:\Program Files\nodejs\node.exe C:\Users\datvt\AppData\Roaming\npm\node_modules\firebase-tools\lib\bin\firebase.js deploy
[debug] [2024-02-02T08:01:25.524Z] CLI Version:   13.1.0
[debug] [2024-02-02T08:01:25.524Z] Platform:      win32
[debug] [2024-02-02T08:01:25.524Z] Node Version:  v20.9.0
[debug] [2024-02-02T08:01:25.525Z] Time:          Fri Feb 02 2024 15:01:25 GMT+0700 (Indochina Time)
[debug] [2024-02-02T08:01:25.525Z] ----------------------------------------------------------------------
[debug] 
[debug] [2024-02-02T08:01:25.529Z] >>> [apiv2][query] GET https://firebase-public.firebaseio.com/cli.json [none]
[debug] [2024-02-02T08:01:25.628Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2024-02-02T08:01:25.628Z] > authorizing via signed-in user (tiendatyenbai2001@gmail.com)
[debug] [2024-02-02T08:01:25.628Z] [iam] checking project dsuit-c23a2 for permissions ["firebase.projects.get","firebasehosting.sites.update"]
[debug] [2024-02-02T08:01:25.629Z] > refreshing access token with scopes: []
[debug] [2024-02-02T08:01:25.629Z] >>> [apiv2][query] POST https://www.googleapis.com/oauth2/v3/token [none]
[debug] [2024-02-02T08:01:25.629Z] >>> [apiv2][body] POST https://www.googleapis.com/oauth2/v3/token [omitted]
[debug] [2024-02-02T08:01:25.638Z] *** [apiv2] error from fetch(https://firebase-public.firebaseio.com/cli.json, {"headers":{},"method":"GET"}): FetchError: request to https://firebase-public.firebaseio.com/cli.json failed, reason: getaddrinfo ENOTFOUND firebase-public.firebaseio.com
[warn] !  Unable to fetch the CLI MOTD and remote config. This is not a fatal error, but may indicate an issue with your network connection. 
[debug] [2024-02-02T08:01:25.640Z] Failed to fetch MOTD FirebaseError: Failed to make request to https://firebase-public.firebaseio.com/cli.json
[debug] [2024-02-02T08:01:25.641Z] *** [apiv2] error from fetch(https://www.googleapis.com/oauth2/v3/token, {"headers":{},"method":"POST","body":{"_overheadLength":545,"_valueLength":212,"_valuesToMeasure":[],"writable":false,"readable":true,"dataSize":0,"maxDataSize":2097152,"pauseStreams":true,"_released":true,"_streams":[],"_currentStream":null,"_insideLoop":false,"_pendingNext":false,"_boundary":"--------------------------080127810923556619719217","_events":{},"_eventsCount":1}}): FetchError: request to https://www.googleapis.com/oauth2/v3/token failed, reason: getaddrinfo ENOTFOUND www.googleapis.com
[debug] [2024-02-02T08:01:25.641Z] [iam] error while checking permissions, command may fail: FirebaseError: Authentication Error: Your credentials are no longer valid. Please run firebase login --reauth

For CI servers and headless environments, generate a new token with firebase login:ci
[debug] [2024-02-02T08:01:25.641Z] > refreshing access token with scopes: []
[debug] [2024-02-02T08:01:25.641Z] >>> [apiv2][query] POST https://www.googleapis.com/oauth2/v3/token [none]
[debug] [2024-02-02T08:01:25.641Z] >>> [apiv2][body] POST https://www.googleapis.com/oauth2/v3/token [omitted]
[debug] [2024-02-02T08:01:25.651Z] *** [apiv2] error from fetch(https://www.googleapis.com/oauth2/v3/token, {"headers":{},"method":"POST","body":{"_overheadLength":545,"_valueLength":212,"_valuesToMeasure":[],"writable":false,"readable":true,"dataSize":0,"maxDataSize":2097152,"pauseStreams":true,"_released":true,"_streams":[],"_currentStream":null,"_insideLoop":false,"_pendingNext":false,"_boundary":"--------------------------955510867251183671582395","_events":{},"_eventsCount":1}}): FetchError: request to https://www.googleapis.com/oauth2/v3/token failed, reason: getaddrinfo ENOTFOUND www.googleapis.com
[debug] [2024-02-02T08:01:25.651Z] Authentication Error: Your credentials are no longer valid. Please run firebase login --reauth

For CI servers and headless environments, generate a new token with firebase login:ci
[error] 
[error] Error: Assertion failed: resolving hosting target of a site with no site name or target name. This should have caused an error earlier
